### ISSUE TITLE
requirements moved to requirements-dev.txt but the latter missing from release tarball.

### ISSUE THREAD

**Issue by limburgher** [boto/botocore#2502]
This breaks RPM builds.

**Comment by nateprewitt** [comment]
Hey @limburgher, thanks for the report! We’re currently in the process of trying to get this fixed. I’ll post and update once we have it resolved.

**Comment by nateprewitt** [comment]
Hey @limburgher, we should have a fix merged shortly in #2503 that will address this. Thanks again for the note!

Out of curiosity, we wanted to better understand what part of the RPM process this broke. `requirements.txt`, and in turn `requirements-dev.txt`, should only be local development components not required for using botocore. Does RPM include these as part of its installation process?

**Comment by limburgher** [comment]
We use the requirements.txt to determine the components needed for build, which for us includes everything in the test suite.

The tests are also failing on this release, but I can't really troubleshoot that further until I know I have the pieces I need in place. :)

**Comment by nateprewitt** [comment]
Ah, so we moved our test suite from nose to pytest in preparation for Python 3.10 which is what led to the requirements-dev.txt rework. I'll leave this open for the day if you have any other feedback/issues to report around that. Otherwise, we'll resolve this in the evening (PT) since the immediate issue is addressed.

**Comment by limburgher** [comment]
I'll just text the next release, assuming it will be soon.

**Comment by nateprewitt** [comment]
Today's release 1.21.50 should now be available.

**Comment by limburgher** [comment]
So it is, let me give it a whirl.

**Comment by limburgher** [comment]
I now have these:

FAILED tests/unit/test_parsers.py::TestTaggedUnions::test_base_xml_parser_handles_unknown_member
FAILED tests/unit/test_parsers.py::TestTaggedUnions::test_base_json_parser_handles_unknwon_member

And for the new awscli, I get:
=========================== short test summary info ============================
ERROR tests/functional/docs/test_examples.py - pkg_resources.DistributionNotF...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 44 warnings, 1 error in 25.21s ========================

**Comment by nateprewitt** [comment]
Curious, neither of the TaggedUnions tests is failing locally or in our CI environments. Is there a specific exception being returned with the failures? Both tests look fairly stateless, so I wouldn't expect environment to have an effect.

For the CLI test, do you have the full exception? We don't invoke pkg_resources directly, so I'm not sure why that would be an error case.

**Comment by limburgher** [comment]
Here, have the logs. :)
[botocore-build.log](https://github.com/boto/botocore/files/7247162/botocore-build.log)
[awscli-build.log](https://github.com/boto/botocore/files/7247163/awscli-build.log)

**Comment by nateprewitt** [comment]
It looks like the Botocore logs are pointing to the system raising multiple warnings. I did a fresh install on Python 3.9 with Fedora 34 with the package and was unable to reproduce. Looking at the test preamble from the logs, it appears this may be installed alongside some conflicting dependencies (`pluggy` specifically). This may be the cause of the additional warnings.

For the CLI, it seems to have a similar issue that `pluggy==1.0.0` is installed but conflicts with our required pytest version of 6.2.5. That's leading to the exception:
```
tests/functional/docs/test_examples.py:28: in <module>
    import docutils.parsers.rst
/usr/lib/python3.9/site-packages/docutils/parsers/rst/__init__.py:75: in <module>
    from docutils.parsers.rst import roles, states
/usr/lib/python3.9/site-packages/docutils/parsers/rst/roles.py:78: in <module>
    from docutils.utils.code_analyzer import Lexer, LexerError
/usr/lib/python3.9/site-packages/docutils/utils/code_analyzer.py:11: in <module>
    from pkg_resources import DistributionNotFound as ResourceError
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:3243: in <module>
    def _initialize_master_working_set():
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:3226: in _call_aside
    f(*args, **kwargs)
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:3255: in _initialize_master_working_set
    working_set = WorkingSet._build_master()
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:570: in _build_master
    return cls._build_from_requirements(__requires__)
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:583: in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
/usr/lib/python3.9/site-packages/pkg_resources/__init__.py:772: in resolve
    raise DistributionNotFound(req, requirers)
E   pkg_resources.DistributionNotFound: The 'pluggy<1.0.0a1,>=0.12' distribution was not found and is required by pytest
```

**Comment by limburgher** [comment]
...which reminds me that I updated my laptop to the rawhide python3-pluggy to test something else. One distro-sync to 0.13 later, and botocore is fine, at least on f34. Testing rawhide.

New cli log:
[awscli-build.log](https://github.com/boto/botocore/files/7247444/awscli-build.log)

**Comment by nateprewitt** [comment]
For the CLI, it looks like you're running one of our integration suite packages. Those all require an AWS account to be [configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) because they're sending requests to active services. I was under the impression most distros didn't run these in their release process. The current failures are expected when credentials aren't configured.

**Comment by limburgher** [comment]
OIC! Eliminating that fixes the lot. Thank you so much!

**Comment by nateprewitt** [comment]
No problem, glad it worked out! I'm going to resolve this for now since we addressed the original packaging concern, and it appears testing for Fedora is relatively unblocked. Please feel free to open a new issue if you encounter more problems. Thanks!

**Comment by limburgher** [comment]
Thank you for your expeditious assistance!
